### PR TITLE
Add Todo list examples

### PR DESCRIPTION
**Pull request by arximboldi**

### CODE REVIEW

File: lager/reader.hpp [hunk 1]
```diff
@@ -42,6 +42,12 @@ struct reader_mixin
         return attred(member, *this);
     }
 
+    template <typename Xform>
+    auto xf(Xform&& xf)
+    {
+        return xform(xf)(*this);
```
Comment (CJBussey) [inline lager/reader.hpp]: It seems you're missing a forward here with `xf` (also in the writer xf impl)

File: lager/extra/qt.hpp [hunk 2]
```diff
@@ -0,0 +1,85 @@
...
+    type LAGER_QT(name);                                                       \
+    Q_PROPERTY(type name READ name CONSTANT)                                   \
+    const type& name() const { return LAGER_QT(name); }                        \
+    /**/
+
+#define LAGER_QT_SIGNAL_DETAIL(type, name)                                     \
+    Q_SIGNAL void name##Changed(const type& name);                             \
+    ::lager::detail::qt_helper funq__##name##__initHelper__ = [this] {         \
+        ::lager::watch(LAGER_QT(name),                                         \
+                       [this](const type& old, const type& curr) {             \
```
Comment (CJBussey) [inline lager/extra/qt.hpp]: `old` will produce a warning an unused variable here

File: lager/extra/qt.hpp [hunk 3]
```diff
@@ -0,0 +1,85 @@
...
+                       });                                                     \
+        return ::lager::detail::qt_helper{};                                   \
+    }();                                                                       \
+    /**/
+
+/*!
+ * Use in a QObject class declaration to define a property supported
+ * by a lager::reader<> value.
+ */
+#define LAGER_QT_READER(type, name)                                            \
```
Comment (CJBussey) [inline lager/extra/qt.hpp]: Very nice :)

File: example/todo/qml/main.cpp [hunk 4]
```diff
@@ -0,0 +1,66 @@
...
+
+public:
+    Todo(lager::cursor<todo> data)
+        : LAGER_QT(done){data[&todo::done]}
+        , LAGER_QT(text){data[&todo::text].xf(
+              zug::map([](auto&& x) { return QString::fromStdString(x); }),
+              zug::map([](auto&& x) { return x.toStdString(); }))}
+    {}
+
+    LAGER_QT_CURSOR(bool, done);
```
Comment (CJBussey) [inline example/todo/qml/main.cpp]: This is very nice :)

File: lager/extra/qt.hpp [hunk 5]
```diff
@@ -0,0 +1,85 @@
...
+
+namespace detail {
+
+struct qt_helper
+{};
+
+} // namespace detail
+
+/*!
+ * Use in a class declaration before any FunQ definition.
```
Comment (CJBussey) [inline lager/extra/qt.hpp]: Is this still true? I can't see you using it anywhere!

Comment (arximboldi) [reply lager/extra/qt.hpp]: Nope :p Good catch!

File: example/todo/qml/main.cpp (hunk 4 above)
Comment (arximboldi) [reply example/todo/qml/main.cpp]: It's ugly to have a macro, but Qt is so verbose...

File: lager/extra/qt.hpp (hunk 2 above)
Comment (arximboldi) [reply lager/extra/qt.hpp]: Not sure I care :p

File: lager/reader.hpp (hunk 1 above)
Comment (arximboldi) [reply lager/reader.hpp]: Good catch!

File: lager/extra/qt.hpp (hunk 2 above)
Comment (CJBussey) [reply lager/extra/qt.hpp]: 😱

File: example/todo/todo.hpp [hunk 6]
```diff
@@ -0,0 +1,39 @@
...
+//
+// This file is part of lager.
+//
+// lager is free software: you can redistribute it and/or modify
+// it under the terms of the MIT License, as detailed in the LICENSE
+// file located at the root of this source code distribution,
+// or here: <https://github.com/arximboldi/lager/blob/master/LICENSE>
+//
+
+#include <immer/flex_vector.hpp>
```
Comment (CJBussey) [inline example/todo/todo.hpp]: This can be removed, and you should add `<string>` and `<vector>`

File: lager/extra/qt.hpp [hunk 7]
```diff
@@ -0,0 +1,85 @@
...
+
+#define LAGER_QT_SIGNAL_DETAIL(type, name)                                     \
+    Q_SIGNAL void name##Changed(const type& name);                             \
+    ::lager::detail::qt_helper funq__##name##__initHelper__ = [this] {         \
+        ::lager::watch(LAGER_QT(name),                                         \
+                       [this](const type& old, const type& curr) {             \
+                           this->name##Changed(curr);                          \
+                       });                                                     \
+        return ::lager::detail::qt_helper{};                                   \
+    }();                                                                       \
```
Comment (CJBussey) [inline lager/extra/qt.hpp]: What do you think about removing the `;` here so the client code is forced to add it and using `LAGER_QT_READER` and `LAGER_QT_CURSOR`?

Comment (arximboldi) [reply lager/extra/qt.hpp]: I thought about that, I am not sure if I prefer ; or not. What do you prefer?

File: example/todo/todo.hpp (hunk 6 above)
Comment (arximboldi) [reply example/todo/todo.hpp]: Good catch :) I actually wanna use `flex_vector`, but need to tweak the `[]` operator of the cursors first.

File: lager/extra/qt.hpp (hunk 7 above)
Comment (CJBussey) [reply lager/extra/qt.hpp]: I did like the idea to keep the syntax as close to a declaration as possible, however, Qt doesn't enforce it with `Q_PROPERTY`, so using `LAGER_QT_READER` and `Q_PROPERTY` side-by-side might feel a bit annoying (having to remember which needs the `;`). But in that case, I would remove the `;` in your usage example.

**Comment by arximboldi** [comment]
The cursor based Todo QML example is reasonably feature complete. I would like to also add some examples based on Redux (one in QML, another in ImGui), but I will do that in later branches. My next goal (in a sepparate branch) is to add some minimal integration between the Redux and cursor parts, basically making the `store` a `reader` cursor, and using that to replace `current()` and the `view()` functions.

**Pull request merged**